src/libraries/System.Private.CoreLib/src/System/Collections/Generic/INonRandomizedEqualityComparer.cs

```diff
@@ -0,0 +1,13 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace System.Collections.Generic
+{
+    internal interface INonRandomizedEqualityComparer<in T> : IEqualityComparer<T>
```

Comment:
Is this interface necessary? Can we just make NonRandomizedStringEqualityComparer unsealed?

Reply:
At the moment it's only for _string_. But this also provides the infrastructure to have it work for other types as well, such as `Utf8String`. (I didn't do that as part of this PR, but it was in one of the early prototypes back in the coreclr repo.)

You recommend deleting it for now and crossing that bridge later?

Reply:
Yes. Future proofing like this works against our goals around framework linkability.